Coding Tasks:

Start a new Jupyter Notebook to complete these tasks. This week, you'll be combining two different datasets.

First, you'll work with a dataset containing the number of primary care physicians per county for each county in the United States. It was obtained from the Area Health Resources File, published by the [Health Resources and Services Administration](https://data.hrsa.gov/topics/health-workforce/ahrf). This data is contained in the file `primary_care_physicians.csv`.

Second, the file `population_by_county.csv` contains the Census Bureau's 2019 population estimates for each US County. It also contains a column `urban`. The `urban` column uses data from the National Bureau of Economic Research to classify each county as either urban or rural. The U.S. Office of Management and Budget designates counties as metropolitan (a core urban area of 50,000 or more population), micropolitan (an urban core of at least 10,000 but less than 50,000 population), or neither. Here, a county is considered "urban" if it is part of a metropolitan or micropolitan area and "rural" if it is not.

 1. First, import the primary care physicians dataset (`primary_care_physicians.csv`) into a data frame named `physicians`. 
 2. Filter `physicians` down to just the counties in Tennessee. Save the filtered dataframe back to `physicians`. Verify that the resulting dataframe has 95 rows.
 3. Look at the distribution of the number of primary care physicians. What do you notice?
 4. Now, import the population by county dataset (`population_by_county.csv`) into a DataFrame named `population`.
 5. Merge the `physicians` DataFrame with the `population` DataFrame. Keep only the values for Tennessee. When you merge, be sure the include both the `population` and `urban` columns in the merged results. Save the result of the merge back to `physicians`.
 6. How many Tennessee counties are considered urban?
 7. The State Health Access Data Assistance Center (SHADAC) (https://www.shadac.org/) classifies counties into three groups based on the number of residents per primary care physician. First, counties with fewer than 1500 residents per primary care physician are considered to have an "adequate" supply. Counties with at least 1500 residents but fewer than 3500 residents per primary care physician are considered to have a "moderately inadequate" supply, and counties with at least 3500 residents per primary care physician are considered to have a "low inadequate" supply. How many counties in Tennessee are in each group? 
 8. Does there appear to be any detectable relationship between whether a county is urban or rural and its supply of primary care physicians?

In [138]:
import pandas as pd 
physicians = pd.read_csv("../data/primary_care_physicians.csv")

In [139]:
physicians.head(5)

FIPS    state   county  primary_care_physicians
0  1001  Alabama  Autauga                     26.0
1  1003  Alabama  Baldwin                    153.0
2  1005  Alabama  Barbour                      8.0
3  1007  Alabama     Bibb                     12.0
4  1009  Alabama   Blount                     12.0

In [140]:
physicians = physicians.loc[physicians['state'] == 'Tennessee']

In [141]:
physicians.shape

(95, 4)

In [142]:
physicians.head(5)

FIPS      state    county  primary_care_physicians
2432  47001  Tennessee  Anderson                     39.0
2433  47003  Tennessee   Bedford                     15.0
2434  47005  Tennessee    Benton                      3.0
2435  47007  Tennessee   Bledsoe                      1.0
2436  47009  Tennessee    Blount                     90.0

In [143]:
physicians['primary_care_physicians'].describe()
#some counties have no physicians, the max is 806 physicians in one county, median is 12 physicians in a county
#The range is super wide, mean should not be used as a measure here. 

count     95.000000
mean      51.042105
std      129.311426
min        0.000000
25%        4.500000
50%       12.000000
75%       26.500000
max      806.000000
Name: primary_care_physicians, dtype: float64

In [144]:
physicians.loc[physicians['primary_care_physicians'] == 806]

FIPS      state  county  primary_care_physicians
2510  47157  Tennessee  Shelby                    806.0

In [145]:
population = pd.read_csv("../data/population_by_county.csv")

In [146]:
population.head(5)

FIPS  population          county     state  urban
0  17051       21565  Fayette County  ILLINOIS  Rural
1  17107       29003    Logan County  ILLINOIS  Rural
2  17165       23994   Saline County  ILLINOIS  Rural
3  17097      701473     Lake County  ILLINOIS  Urban
4  17127       14219   Massac County  ILLINOIS  Rural

In [147]:
physicians['state'] = physicians['state'].str.upper()

In [148]:
physicians.head(5)

FIPS      state    county  primary_care_physicians
2432  47001  TENNESSEE  Anderson                     39.0
2433  47003  TENNESSEE   Bedford                     15.0
2434  47005  TENNESSEE    Benton                      3.0
2435  47007  TENNESSEE   Bledsoe                      1.0
2436  47009  TENNESSEE    Blount                     90.0

In [149]:
population = population.loc[population['state'] == 'TENNESSEE']

In [150]:
population.shape

(95, 5)

In [151]:
population.head(5)

FIPS  population            county      state  urban
283  47165      183437     Sumner County  TENNESSEE  Urban
284  47169       10231  Trousdale County  TENNESSEE  Urban
285  47027        7654       Clay County  TENNESSEE  Rural
405  47157      936374     Shelby County  TENNESSEE  Urban
406  47077       27977  Henderson County  TENNESSEE  Rural

In [152]:
#population['county'] = population['county'].str.replace('County','')

In [153]:
population.head(10)

FIPS  population             county      state  urban
283  47165      183437      Sumner County  TENNESSEE  Urban
284  47169       10231   Trousdale County  TENNESSEE  Urban
285  47027        7654        Clay County  TENNESSEE  Rural
405  47157      936374      Shelby County  TENNESSEE  Urban
406  47077       27977   Henderson County  TENNESSEE  Rural
407  47089       53679   Jefferson County  TENNESSEE  Urban
408  47153       14816  Sequatchie County  TENNESSEE  Urban
409  47017       27886     Carroll County  TENNESSEE  Rural
410  47037      687488    Davidson County  TENNESSEE  Urban
411  47135        7962       Perry County  TENNESSEE  Rural

In [154]:
population.tail(10)

FIPS  population           county      state  urban
3190  47063       64132   Hamblen County  TENNESSEE  Urban
3191  47173       19488     Union County  TENNESSEE  Urban
3192  47071       25715    Hardin County  TENNESSEE  Rural
3193  47043       52680   Dickson County  TENNESSEE  Urban
3194  47129       21545    Morgan County  TENNESSEE  Rural
3195  47123       46064    Monroe County  TENNESSEE  Rural
3196  47079       32284     Henry County  TENNESSEE  Rural
3197  47033       14399  Crockett County  TENNESSEE  Rural
3198  47095        7401      Lake County  TENNESSEE  Rural
3199  47093      461104      Knox County  TENNESSEE  Urban

In [155]:
physicians.shape

(95, 4)

In [156]:
physicians.head(10)

FIPS      state    county  primary_care_physicians
2432  47001  TENNESSEE  Anderson                     39.0
2433  47003  TENNESSEE   Bedford                     15.0
2434  47005  TENNESSEE    Benton                      3.0
2435  47007  TENNESSEE   Bledsoe                      1.0
2436  47009  TENNESSEE    Blount                     90.0
2437  47011  TENNESSEE   Bradley                     55.0
2438  47013  TENNESSEE  Campbell                     19.0
2439  47015  TENNESSEE    Cannon                      3.0
2440  47017  TENNESSEE   Carroll                     12.0
2441  47019  TENNESSEE    Carter                     22.0

In [157]:
physicians.tail(10)

FIPS      state      county  primary_care_physicians
2517  47171  TENNESSEE      Unicoi                      6.0
2518  47173  TENNESSEE       Union                      3.0
2519  47175  TENNESSEE   Van Buren                      0.0
2520  47177  TENNESSEE      Warren                     21.0
2521  47179  TENNESSEE  Washington                    226.0
2522  47181  TENNESSEE       Wayne                      5.0
2523  47183  TENNESSEE     Weakley                     18.0
2524  47185  TENNESSEE       White                      9.0
2525  47187  TENNESSEE  Williamson                    338.0
2526  47189  TENNESSEE      Wilson                     43.0

In [158]:
physicians = pd.merge(left = physicians, right = population, left_on = 'FIPS', right_on = 'FIPS')

In [159]:
physicians.shape

(95, 8)

In [160]:
physicians

FIPS    state_x    county_x  primary_care_physicians  population  \
0   47001  TENNESSEE    Anderson                     39.0       76061   
1   47003  TENNESSEE     Bedford                     15.0       48292   
2   47005  TENNESSEE      Benton                      3.0       16140   
3   47007  TENNESSEE     Bledsoe                      1.0       14836   
4   47009  TENNESSEE      Blount                     90.0      129927   
..    ...        ...         ...                      ...         ...   
90  47181  TENNESSEE       Wayne                      5.0       16693   
91  47183  TENNESSEE     Weakley                     18.0       33510   
92  47185  TENNESSEE       White                      9.0       26800   
93  47187  TENNESSEE  Williamson                    338.0      225389   
94  47189  TENNESSEE      Wilson                     43.0      136666   

             county_y    state_y  urban  
0     Anderson County  TENNESSEE  Urban  
1      Bedford County  TENNESSEE  Rural  
2       Benton County  TENNESSEE  Rural  
3      Bledsoe County  TENNESSEE  Rural  
4       Blount County  TENNESSEE  Urban  
..                ...        ...    ...  
90       Wayne County  TENNESSEE  Rural  
91     Weakley County  TENNESSEE  Rural  
92       White County  TENNESSEE  Rural  
93  Williamson County  TENNESSEE  Urban  
94      Wilson County  TENNESSEE  Urban  

[95 rows x 8 columns]

In [161]:
physicians.loc[physicians['urban'] == 'Urban'].shape
#38 urban counties, 

(38, 8)

In [162]:
physicians.loc[physicians['urban'] == 'Rural'].shape
#57 rural counties

(57, 8)

In [164]:
physicians["pop_per_physician"] = physicians['population']/physicians['primary_care_physicians']

In [165]:
physicians.head(5)

FIPS    state_x  county_x  primary_care_physicians  population  \
0  47001  TENNESSEE  Anderson                     39.0       76061   
1  47003  TENNESSEE   Bedford                     15.0       48292   
2  47005  TENNESSEE    Benton                      3.0       16140   
3  47007  TENNESSEE   Bledsoe                      1.0       14836   
4  47009  TENNESSEE    Blount                     90.0      129927   

          county_y    state_y  urban  pop_per_physician  
0  Anderson County  TENNESSEE  Urban        1950.282051  
1   Bedford County  TENNESSEE  Rural        3219.466667  
2    Benton County  TENNESSEE  Rural        5380.000000  
3   Bledsoe County  TENNESSEE  Rural       14836.000000  
4    Blount County  TENNESSEE  Urban        1443.633333

In [215]:
adequacy = [] 
for i in physicians['pop_per_physician'] :
    if i < 1500 :
        adequacy.append("adequate")
    elif 1500 <= i < 3500 :
        adequacy.append("moderately inadequate")
    elif i >= 3500 :
        adequacy.append("low inadequate")
    else :
        adequacy.append("No information")
    


In [216]:
physicians["adequacy"] = adequacy

In [217]:
physicians.head(10)

FIPS    state_x  county_x  primary_care_physicians  population  \
0  47001  TENNESSEE  Anderson                     39.0       76061   
1  47003  TENNESSEE   Bedford                     15.0       48292   
2  47005  TENNESSEE    Benton                      3.0       16140   
3  47007  TENNESSEE   Bledsoe                      1.0       14836   
4  47009  TENNESSEE    Blount                     90.0      129927   
5  47011  TENNESSEE   Bradley                     55.0      105749   
6  47013  TENNESSEE  Campbell                     19.0       39797   
7  47015  TENNESSEE    Cannon                      3.0       14178   
8  47017  TENNESSEE   Carroll                     12.0       27886   
9  47019  TENNESSEE    Carter                     22.0       56433   

          county_y    state_y  urban  pop_per_physician               adequacy  
0  Anderson County  TENNESSEE  Urban        1950.282051  moderately inadequate  
1   Bedford County  TENNESSEE  Rural        3219.466667  moderately inadequate  
2    Benton County  TENNESSEE  Rural        5380.000000         low inadequate  
3   Bledsoe County  TENNESSEE  Rural       14836.000000         low inadequate  
4    Blount County  TENNESSEE  Urban        1443.633333               adequate  
5   Bradley County  TENNESSEE  Urban        1922.709091  moderately inadequate  
6  Campbell County  TENNESSEE  Rural        2094.578947  moderately inadequate  
7    Cannon County  TENNESSEE  Urban        4726.000000         low inadequate  
8   Carroll County  TENNESSEE  Rural        2323.833333  moderately inadequate  
9    Carter County  TENNESSEE  Urban        2565.136364  moderately inadequate

In [218]:
physicians.loc[physicians['adequacy'] == 'moderately inadequate'].shape
#50 counties

(50, 10)

In [219]:
physicians.loc[physicians['adequacy'] == 'No information'].shape

(0, 10)

In [220]:
physicians.loc[physicians['adequacy'] == 'adequate'].shape
#14 counties 

(14, 10)

In [221]:
physicians.loc[physicians['adequacy'] == 'low inadequate'].shape
#31 counties 

(31, 10)

In [229]:
physicians.loc[physicians['urban'] == 'Urban']['primary_care_physicians'].describe()

count     38.000000
mean     108.315789
std      191.346736
min        1.000000
25%        6.500000
50%       22.500000
75%       90.750000
max      806.000000
Name: primary_care_physicians, dtype: float64

In [230]:
physicians.loc[physicians['urban'] == 'Rural']['primary_care_physicians'].describe()

count    57.000000
mean     12.859649
std      12.930361
min       0.000000
25%       3.000000
50%       9.000000
75%      18.000000
max      53.000000
Name: primary_care_physicians, dtype: float64

In [231]:
Urban = physicians.loc[physicians['urban'] == 'Urban']

In [232]:
Urban['primary_care_physicians'].describe()

count     38.000000
mean     108.315789
std      191.346736
min        1.000000
25%        6.500000
50%       22.500000
75%       90.750000
max      806.000000
Name: primary_care_physicians, dtype: float64

In [235]:
Urban.loc[Urban['primary_care_physicians'] == 1]

FIPS    state_x county_x  primary_care_physicians  population  \
80  47161  TENNESSEE  Stewart                      1.0       13427   

          county_y    state_y  urban  pop_per_physician        adequacy  
80  Stewart County  TENNESSEE  Urban            13427.0  low inadequate

In [242]:
Urban.loc[Urban['primary_care_physicians'] == 806]

FIPS    state_x county_x  primary_care_physicians  population  \
78  47157  TENNESSEE   Shelby                    806.0      936374   

         county_y    state_y  urban  pop_per_physician  adequacy  
78  Shelby County  TENNESSEE  Urban        1161.754342  adequate

In [233]:
Rural = physicians.loc[physicians['urban'] == 'Rural']

In [234]:
Rural['primary_care_physicians'].describe()

count    57.000000
mean     12.859649
std      12.930361
min       0.000000
25%       3.000000
50%       9.000000
75%      18.000000
max      53.000000
Name: primary_care_physicians, dtype: float64

In [237]:
Rural.loc[Rural['primary_care_physicians'] == 0]

FIPS    state_x   county_x  primary_care_physicians  population  \
16  47033  TENNESSEE   Crockett                      0.0       14399   
30  47061  TENNESSEE     Grundy                      0.0       13344   
47  47095  TENNESSEE       Lake                      0.0        7401   
87  47175  TENNESSEE  Van Buren                      0.0        5760   

            county_y    state_y  urban  pop_per_physician        adequacy  
16   Crockett County  TENNESSEE  Rural                inf  low inadequate  
30     Grundy County  TENNESSEE  Rural                inf  low inadequate  
47       Lake County  TENNESSEE  Rural                inf  low inadequate  
87  Van Buren County  TENNESSEE  Rural                inf  low inadequate

In [240]:
Rural.loc[Rural['adequacy'] == 'low inadequate'].shape

(21, 10)

In [241]:
Urban.loc[Urban['adequacy'] == 'low inadequate'].shape

(10, 10)

In [244]:
Rural.loc[Rural['adequacy'] == 'adequate'].shape

(5, 10)

In [245]:
Urban.loc[Urban['adequacy'] == 'adequate'].shape

(9, 10)

In [246]:
Rural.loc[Rural['adequacy'] == 'moderately inadequate'].shape

(31, 10)

In [247]:
Urban.loc[Urban['adequacy'] == 'moderately inadequate'].shape

(19, 10)

Cursorily yes, it looks like the rural counties tend to have fewer physicians, though there are more rural (57) than urban (38) in Tennessee. Rural mean = 12, median 9; Urban mean = 108, median = 22.5 Further the number of counties with low-inadequacy also differ - with rural counties having 21 counties with low inadequacy (37%) as compared to urban counties (26%), adequate supply in rural counties (9%) as compared to urban (24%), moderately adequate supply in rural (54%) as compared to urban (50%). However, to prove this you should run a log regression! 

In [1]:
population['county'].str.rsplit(' ', n = 1, expand = True)

NameError: name 'population' is not defined

In [ ]:
population['county']=population['county'].str.split(' County', expand = True)[0]
#add the space in front of the County to take out the space! 